### PR TITLE
✨PIC-1075 custom event for update preferences

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**
Not sure how to verify that the app insights event actually goes until deployment to dev .

Anyways, it's much the same as court case service. Assuming all the wiring and app insights set up is done. Certainly looks like it.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/config/TelemetryConfig.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      return StringUtils.isEmpty(context.environment.getProperty("application.insights.ikey"))
+    }
+  }
+
+  @Bean
+  @Profile("!test")
+  @RequestScope
+  fun requestProperties(): Map<String, String> {
+    return Optional.ofNullable(ThreadContext.getRequestTelemetryContext())
+      .map { obj: RequestTelemetryContext -> obj.httpRequestTelemetry }
```
Comment (chris-faulkner-mojdt) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/config/TelemetryConfig.kt:42]: I'm sure there is a more idiomatic way of doing this in Kotlin.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/service/TelemetryServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@ExtendWith(MockitoExtension::class)
+internal class TelemetryServiceTest {
+  @Mock
+  lateinit var telemetryClient: TelemetryClient
+
+  @InjectMocks
+  lateinit var telemetryService: TelemetryService
+
+  @Test
+  fun `Put preferences should store preferences and return DTO`() {
```
Comment (chris-faulkner-mojdt) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/service/TelemetryServiceTest.kt:20]: Seems odd that there's only one test but Kotlin guarantees that all of the parameters are non-null which covers the edge cases. I think.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/config/TelemetryConfig.kt (hunk 1 above)
Comment (jhevans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/config/TelemetryConfig.kt:42]: Can you use the [Elvis operator](https://kotlinlang.org/docs/null-safety.html#elvis-operator)?
```
ThreadContext.getRequestTelemetryContext()?
. map { ... }
?: emptyMap()
```

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/service/TelemetryServiceTest.kt (hunk 2 above)
Comment (jhevans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuserpreferences/service/TelemetryServiceTest.kt:20]: yeah... seems pretty robust! Kotlin did good 😄

**Review by jhevans — APPROVED** [review]
Looks good, one possible suggestion in response to your comment but don't know if it works or not!
